internal/system/snapd_client.go

```diff
@@ -0,0 +1,155 @@
+package system
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+const (
+	// StatusActive represents an active snap installation.
+	StatusActive = "active"
```

Comment:
Another reason this should be in a separate package, is because `StatusActive` in the "system" package doesn't indicate that it's related to snapd. But if in a separate package it'd be `snapd.StatusActive`.